### PR TITLE
Add role memberships command

### PR DESCRIPTION
**Pull request by gl-johnson**
### Desired Outcome

Add support for the 'role memberships' command that previously existing in the Ruby CLI.

### Implemented Changes

Add CLI functionality that allows commands with the syntax `conjur role memberships -i user:someuser`.
Support both recursive (default) and direct membership queries (-d / --direct). Drop support for the --system flag which served no purpose.

### Connected Issue/Story

CyberArk internal issue link: [ONYX-23087](https://ca-il-jira.il.cyber-ark.com:8443/browse/ONYX-23087)

### Definition of Done
*At least 1 todo must be completed in the sections below for the PR to be
merged.*

#### Changelog

- [x] The CHANGELOG has been updated, or
- [ ] This PR does not include user-facing changes and doesn't require a
  CHANGELOG update

#### Test coverage

- [x] This PR includes new unit and integration tests to go with the code
  changes, or
- [ ] The changes in this PR do not require tests

#### Documentation

- [ ] Docs (e.g. `README`s) were updated in this PR
- [x] A follow-up issue to update official docs has been filed here: [ONYX-22999](https://ca-il-jira.il.cyber-ark.com:8443/browse/ONYX-22999)
- [ ] This PR does not require updating any documentation

#### Behavior

- [ ] This PR changes product behavior and has been reviewed by a PO, or
- [x] These changes are part of a larger initiative that will be reviewed later, or
- [ ] No behavior was changed with this PR

#### Security

- [ ] Security architect has reviewed the changes in this PR,
- [x] These changes are part of a larger initiative with a separate security review, or
- [ ] There are no security aspects to these changes

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,8 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 ## [7.2.0] - 2022-08-02
 
 ### Added
+- Add "role memberships" subcommand
+  [cyberark/cyberark-conjur-cli#417](https://github.com/cyberark/cyberark-conjur-cli/pull/419)
 - Add "role" command and "exists" subcommand
   [cyberark/cyberark-conjur-cli#417](https://github.com/cyberark/cyberark-conjur-cli/pull/417)
```
Comment (szh) [inline CHANGELOG.md:15]: I wonder if we should combine these into one item

File: test/test_integration_role.py [hunk 2]
```diff
@@ -129,6 +129,84 @@ def test_role_exists_policy_returns_result(self):
...
+        output = self.invoke_cli(self.cli_auth_params,
+                                 ['role', 'memberships', '-i', 'group:somegroup'])
+        self.assertIn('group:somegroup', output)
+
+    @integration_test(True)
+    def test_role_memberships_policy_returns_result(self):
+        output = self.invoke_cli(self.cli_auth_params,
+                                 ['role', 'memberships', '-i', 'policy:root'])
+        self.assertIn('policy:root', output)
+
```
Comment (szh) [inline test/test_integration_role.py:203]: Were you planning on adding more integration tests for things like direct vs. indirect memberships? I think that would be valuable given that the API repo doesn't have in depth integration tests.

Comment (gl-johnson) [reply test/test_integration_role.py:203]: Good thought - pushed up a few more tests to validate the memberships returned.

Comment (szh) [reply test/test_integration_role.py:203]: Great. I approved https://github.com/cyberark/conjur-api-python/pull/33 and will finish reviewing this one now.

**Review by szh — APPROVED** [review]
LGTM!

**Review by john-odonnell — APPROVED** [review]
LGTM 👍

**Pull request merged**